source/GameContainer.js

```diff
@@ -4,18 +4,29 @@ import Junkership from "./Junkership.js"
 import Trashbot from "./Trashbot.js"
 import Reference from "./Reference.js"
 import Textures from "./Textures.js"
+import Junk from "./Junk.js"
 
 export default class GameContainer extends Pixi.Container {
     constructor() {
         super()
         this.playerCount = 0
+        this.countdownToJunk = 5
         Textures.initTex()
     }
+
     gameOver() {
         console.log("Respawning Junkership")
         game.addChild(new Junkership())
     }
+
     spawnWave() {
-        this.addChild(new Trashbot(Reference.GAME_WIDTH, Reference.GAME_HEIGHT/2))
+        this.addChild(new Trashbot(Reference.GAME_WIDTH, Reference.GAME_HEIGHT / 2))
+    }
+
+    untilJunk(x, y) {
+        if (this.countdownToJunk == 0) {
+            this.countdownToJunk = Math.floor((Math.random() * 5) + 6) // Random number between 5-10
+            this.addChild(new Junk(x, y))
+        } else this.countdownToJunk--
     }
-}
+}
```

Comment:
Are you moving the entire expression to Reference or just the range (i.e. values of 5 and 10)? If you put the expression in the Reference class, it will only be called once, but if you put the range values in should remain random. Probably the best way to define it in Reference is something like `module.exports.JUNK_FREQUENCY_RANGE = { lower: 5, upper: 10 }` then use the values of `Reference.JUNK_FREQUENCY_RANGE.lower` and `Reference.JUNK_FREQUENCY_RANGE.upper` in the expression that's still defined in the GameContainer class. Also, my expression was a little off. It would only return values between 6 and 10, but if you use `Reference.JUNK_FREQUENCY_RANGE.upper - Math.floor(Math.random() * (Reference.JUNK_FREQUENCY_RANGE.lower + 1)) ` the whole thing should work as expected.